### ISSUE TITLE
Multiple streams conflict with each other

### ISSUE THREAD

**Issue by johnnysprinkles** [rogerc/file-stream-rotator#45]
I have multiple streams, boiled down to an example:

```
var stream1 = fsr.getStream({
  date_format: 'YYYY-MM-DD-HH-mm-ss',
  filename: logDir + '/stream1.%DATE%',
  frequency: 'custom',
  verbose: false,
  max_logs: 2,
  audit_file: logDir + '/.audit1.json'
});
var stream2 = fsr.getStream({
  date_format: 'YYYY-MM-DD-HH-mm-ss',
  filename: logDir + '/stream2.%DATE%',
  frequency: 'custom',
  verbose: false,
  max_logs: 2,
  audit_file: logDir + '/.audit2.json'
});

app.use((req, res, next) => {
  stream1.write('x\n');
  stream2.write('x\n');
  next();
})
```

Initially I had it without the "audit_file" but they were both sharing the .audit.json and seemed to not each be getting their full "max_logs" allotment. So then I tried adding separate audit_file locations, but it had no effect it only create one.

Is this supposed to work with multiple streams in the same application? In my case I'd have separate streams for access logs and error logs.

**Comment by johnnysprinkles** [comment]
The seconds in the datestamp is for testing of course. Would normally be hourly.

**Comment by johnnysprinkles** [comment]
My initial impression is that log removal is a bit experimental, and I might want to do it myself instead. Would this be a fair assessment?

**Comment by rogerc** [comment]
Hi @johnnysprinkles  I've released v0.4.0 that fixes the issue. Now each stream instance will use its own auditLog file. Thanks for raising the issue. Let me know if still not working as you'd expect.

**Comment by johnnysprinkles** [comment]
Thanks! Quick turnaround, I'll try it out.
